src/authentication/steps/SelectServer.jsx

```diff
@@ -21,81 +22,92 @@ export class SelectServer extends Component {
   }
 
   componentDidMount() {
-    this.serverInput.focus()
+    this.input.focus()
   }
 
   componentWillReceiveProps(nextProps) {
-    this.setState({
+    this.setState(state => ({
+      ...state,
       error: nextProps.externalError ? ERR_WRONG_ADDRESS : null,
       fetching: nextProps.fetching
-    })
+    }))
   }
 
   componentDidUpdate() {
     if (this.state.error) {
-      this.serverInput.focus()
-      this.serverInput.select()
+      this.input.focus()
+      this.input.select()
     }
   }
 
-  validateValue(e) {
-    this.setState({ value: e.target.value, error: null })
+  onChange(value) {
+    this.setState(state => ({ ...state, error: null }))
+    this.setState(state => ({ ...state, value: value.trim() }))
```

Comment:
Why do you `setState` twice here?

Reply:
Because this is more or less uncoupled.
In fact, setting error state to null should not be done `onChange` but `onBlur` or `onEnter`.
In the other hand, the `trim()` function may occur on every key press because we don't want space character at all.

So... I will move the setting error state to null in another event handler. Thanks for your comment.